NoSuchMethodError when opening AppSec Kit UI in Vaadin 23 apps

There is a `NoSuchMethodError` for [`ComboBox::setItems()`](https://github.com/vaadin/appsec-kit/blob/2.0/appsec-kit-starter/src/main/java/com/vaadin/appsec/views/VulnerabilitiesView.java#L124) when opening the AppSec Kit UI in Vaadin 23 applications using the `appsec-kit-starter:2.0.0.alpha1` dependency.

The cause of the issue is that in `2.0.0.alpha1` the `vaadin-combo-box-flow:14.10.10` -> `flow-data:2.9.10` is used where the [`HasItems::setItems()`](https://vaadin.com/api/platform/14.10.10/com/vaadin/flow/data/binder/HasItems.html#setItems-T...-) method is `void` and it is different from the corresponding one used in `vaadin-combo-box-flow:23.3.28` -> `flow-data:23.3.24` where the [`HasListDataView::setItems()`](https://vaadin.com/api/platform/23.3.24/com/vaadin/flow/data/provider/HasListDataView.html#setItems(T...)) method has a return value.

A possible solution could be to use the `ComboBox::setDataProvider()` method to populate the data which is available both in V14 and V23 versions but the [V23 upgrade guide](https://vaadin.com/docs/v23/upgrading/recommended-changes#replace-deprecated-apis) suggests to use the `setItems()` methods instead.

Affected versions: 2.0.0

```
There was an exception while trying to navigate to 'vaadin-appsec-kit' with the root cause 'java.lang.NoSuchMethodError:
'void com.vaadin.flow.component.combobox.ComboBox.setItems(java.lang.Object[])''
org.springframework.beans.factory.BeanCreationException: Error creating bean with name
'com.vaadin.appsec.views.AppSecView': Instantiation of bean failed; nested exception is
org.springframework.beans.BeanInstantiationException: Failed to instantiate [com.vaadin.appsec.views.AppSecView]:
Constructor threw exception; nested exception is java.lang.NoSuchMethodError: 'void
com.vaadin.flow.component.combobox.ComboBox.setItems(java.lang.Object[])'
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.instantiateBean(AbstractAutowireCapableBeanFactory.java:1334)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBeanInstance(AbstractAutowireCapableBeanFactory.java:1232)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:582)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:542)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:332)
	at com.vaadin.flow.spring.SpringInstantiator.getOrCreate(SpringInstantiator.java:121)
	at com.vaadin.flow.di.Instantiator.createRouteTarget(Instantiator.java:193)
	at com.vaadin.flow.router.internal.AbstractNavigationStateRenderer.lambda$getRouteTarget$1(AbstractNavigationStateRenderer.java:133)
	at java.base/java.util.Optional.orElseGet(Optional.java:364)
	at com.vaadin.flow.router.internal.AbstractNavigationStateRenderer.getRouteTarget(AbstractNavigationStateRenderer.java:132)
	at com.vaadin.flow.router.internal.AbstractNavigationStateRenderer.sendBeforeEnterEventAndPopulateChain(AbstractNavigationStateRenderer.java:493)
	at com.vaadin.flow.router.internal.AbstractNavigationStateRenderer.createChainIfEmptyAndExecuteBeforeEnterNavigation(AbstractNavigationStateRenderer.java:474)
	at com.vaadin.flow.router.internal.AbstractNavigationStateRenderer.handle(AbstractNavigationStateRenderer.java:212)
	at com.vaadin.flow.component.internal.JavaScriptNavigationStateRenderer.handle(JavaScriptNavigationStateRenderer.java:78)
	at com.vaadin.flow.component.internal.JavaScriptBootstrapUI.handleNavigation(JavaScriptBootstrapUI.java:322)
	at com.vaadin.flow.component.internal.JavaScriptBootstrapUI.renderViewForRoute(JavaScriptBootstrapUI.java:285)
	at com.vaadin.flow.component.internal.JavaScriptBootstrapUI.connectClient(JavaScriptBootstrapUI.java:151)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
	at java.base/java.lang.reflect.Method.invoke(Method.java:578)
	at com.vaadin.flow.server.communication.rpc.PublishedServerEventHandlerRpcHandler.invokeMethod(PublishedServerEventHandlerRpcHandler.java:227)
	at com.vaadin.flow.server.communication.rpc.PublishedServerEventHandlerRpcHandler.invokeMethod(PublishedServerEventHandlerRpcHandler.java:204)
	at com.vaadin.flow.server.communication.rpc.PublishedServerEventHandlerRpcHandler.invokeMethod(PublishedServerEventHandlerRpcHandler.java:150)
	at com.vaadin.flow.server.communication.rpc.PublishedServerEventHandlerRpcHandler.handleNode(PublishedServerEventHandlerRpcHandler.java:133)
	at com.vaadin.flow.server.communication.rpc.AbstractRpcInvocationHandler.handle(AbstractRpcInvocationHandler.java:75)
	at com.vaadin.flow.server.communication.ServerRpcHandler.handleInvocationData(ServerRpcHandler.java:438)
	at com.vaadin.flow.server.communication.ServerRpcHandler.lambda$handleInvocations$1(ServerRpcHandler.java:419)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
	at com.vaadin.flow.server.communication.ServerRpcHandler.handleInvocations(ServerRpcHandler.java:419)
	at com.vaadin.flow.server.communication.ServerRpcHandler.handleRpc(ServerRpcHandler.java:320)
	at com.vaadin.flow.server.communication.UidlRequestHandler.synchronizedHandleRequest(UidlRequestHandler.java:115)
	at com.vaadin.flow.server.SynchronizedRequestHandler.handleRequest(SynchronizedRequestHandler.java:40)
	at com.vaadin.flow.server.VaadinService.handleRequest(VaadinService.java:1560)
	at com.vaadin.flow.server.VaadinServlet.service(VaadinServlet.java:369)
	at com.vaadin.flow.spring.SpringServlet.service(SpringServlet.java:106)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:623)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:209)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:153)
	at org.apache.catalina.core.ApplicationDispatcher.invoke(ApplicationDispatcher.java:661)
	at org.apache.catalina.core.ApplicationDispatcher.processRequest(ApplicationDispatcher.java:425)
	at org.apache.catalina.core.ApplicationDispatcher.doForward(ApplicationDispatcher.java:330)
	at org.apache.catalina.core.ApplicationDispatcher.forward(ApplicationDispatcher.java:294)
	at org.springframework.web.servlet.mvc.ServletForwardingController.handleRequestInternal(ServletForwardingController.java:141)
	at org.springframework.web.servlet.mvc.AbstractController.handleRequest(AbstractController.java:177)
	at org.springframework.web.servlet.mvc.SimpleControllerHandlerAdapter.handle(SimpleControllerHandlerAdapter.java:51)
	at org.springframework.web.servlet.DispatcherServlet.doDispatch(DispatcherServlet.java:1072)
	at org.springframework.web.servlet.DispatcherServlet.doService(DispatcherServlet.java:965)
	at org.springframework.web.servlet.FrameworkServlet.processRequest(FrameworkServlet.java:1006)
	at org.springframework.web.servlet.FrameworkServlet.doPost(FrameworkServlet.java:909)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:555)
	at org.springframework.web.servlet.FrameworkServlet.service(FrameworkServlet.java:883)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:623)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:209)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:153)
	at org.apache.tomcat.websocket.server.WsFilter.doFilter(WsFilter.java:51)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:178)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:153)
	at org.springframework.web.filter.RequestContextFilter.doFilterInternal(RequestContextFilter.java:100)
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:117)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:178)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:153)
	at org.springframework.web.filter.FormContentFilter.doFilterInternal(FormContentFilter.java:93)
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:117)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:178)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:153)
	at org.springframework.web.filter.CharacterEncodingFilter.doFilterInternal(CharacterEncodingFilter.java:201)
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:117)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:178)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:153)
	at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:168)
	at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:90)
	at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:481)
	at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:130)
	at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:93)
	at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:74)
	at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:342)
	at org.apache.coyote.http11.Http11Processor.service(Http11Processor.java:390)
	at org.apache.coyote.AbstractProcessorLight.process(AbstractProcessorLight.java:63)
	at org.apache.coyote.AbstractProtocol$ConnectionHandler.process(AbstractProtocol.java:928)
	at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1794)
	at org.apache.tomcat.util.net.SocketProcessorBase.run(SocketProcessorBase.java:52)
	at org.apache.tomcat.util.threads.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1191)
	at org.apache.tomcat.util.threads.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:659)
	at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61)
	at java.base/java.lang.Thread.run(Thread.java:1589)
Caused by: org.springframework.beans.BeanInstantiationException: Failed to instantiate [com.vaadin.appsec.views.AppSecView]: Constructor threw exception; nested exception is java.lang.NoSuchMethodError: 'void com.vaadin.flow.component.combobox.ComboBox.setItems(java.lang.Object[])'
	at org.springframework.beans.BeanUtils.instantiateClass(BeanUtils.java:224)
	at org.springframework.beans.factory.support.SimpleInstantiationStrategy.instantiate(SimpleInstantiationStrategy.java:87)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.instantiateBean(AbstractAutowireCapableBeanFactory.java:1326)
	... 83 more
Caused by: java.lang.NoSuchMethodError: 'void com.vaadin.flow.component.combobox.ComboBox.setItems(java.lang.Object[])'
	at com.vaadin.appsec.views.VulnerabilitiesView.buildFilters(VulnerabilitiesView.java:124)
	at com.vaadin.appsec.views.VulnerabilitiesView.<init>(VulnerabilitiesView.java:46)
	at com.vaadin.appsec.views.AppSecView.buildTabs(AppSecView.java:114)
	at com.vaadin.appsec.views.AppSecView.buildLayout(AppSecView.java:73)
	at com.vaadin.appsec.views.AppSecView.<init>(AppSecView.java:65)
	at java.base/jdk.internal.reflect.DirectConstructorHandleAccessor.newInstance(DirectConstructorHandleAccessor.java:67)
	at java.base/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:500)
	at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:484)
	at org.springframework.beans.BeanUtils.instantiateClass(BeanUtils.java:211)
	... 85 more
Online


```

--- a/appsec-kit-starter/src/main/java/com/vaadin/appsec/views/DependenciesView.java
+++ b/appsec-kit-starter/src/main/java/com/vaadin/appsec/views/DependenciesView.java
@@ -8,6 +8,7 @@
  */
 package com.vaadin.appsec.views;
 
+import java.util.Arrays;
 import java.util.List;
 import java.util.Objects;
 import java.util.Set;
@@ -98,12 +99,9 @@ public class DependenciesView extends AbstractAppSecView {
     public void refresh() {
         Set<Dependency> selectedItems = grid.getSelectedItems();
         grid.deselectAll();
-        grid.setItems(AppSecService.getInstance().getDependencies());
+        grid.setDataProvider(getDependencyDataProvider());
 
-        List<String> sortedGroups = getListDataProvider().getItems().stream()
-                .map(Dependency::getGroup).filter(Objects::nonNull).distinct()
-                .sorted().collect(Collectors.toList());
-        group.setItems(sortedGroups);
+        group.setDataProvider(getGroupDataProvider());
         applyFilters();
         selectedItems.forEach(grid::select);
     }
@@ -115,7 +113,7 @@ public class DependenciesView extends AbstractAppSecView {
 
     private void buildFilters() {
         ecosystem = new ComboBox<>("Ecosystem");
-        ecosystem.setItems(Ecosystem.MAVEN, Ecosystem.NPM);
+        ecosystem.setDataProvider(getEcosystemDataProvider());
         ecosystem.addValueChangeListener(event -> applyFilters());
 
         group = new ComboBox<>("Dependency group");
@@ -124,19 +122,16 @@ public class DependenciesView extends AbstractAppSecView {
 
         if (includeNpmDevDeps) {
             isDevelopment = new ComboBox<>("Is development?");
-            isDevelopment.setItems(Boolean.TRUE, Boolean.FALSE);
+            isDevelopment.setDataProvider(getIsDevelopmentDataProvider());
             isDevelopment.addValueChangeListener(event -> applyFilters());
         }
 
         severity = new ComboBox<>("Severity");
-        severity.setItems(SeverityLevel.NONE, SeverityLevel.LOW,
-                SeverityLevel.MEDIUM, SeverityLevel.HIGH,
-                SeverityLevel.CRITICAL);
+        severity.setDataProvider(getSeverityLevelDataProvider());
         severity.addValueChangeListener(event -> applyFilters());
 
         riskScore = new ComboBox<>("CVSS score");
-        riskScore.setItems(">=0", ">=1", ">=2", ">=3", ">=4", ">=5", ">=6",
-                ">=7", ">=8", ">=9", "=10");
+        riskScore.setDataProvider(getRiskScoreDataProvider());
         riskScore.addValueChangeListener(event -> applyFilters());
 
         List<Component> components = Stream
@@ -212,5 +207,34 @@ public class DependenciesView extends AbstractAppSecView {
     @SuppressWarnings("unchecked")
     private ListDataProvider<Dependency> getListDataProvider() {
         return (ListDataProvider<Dependency>) grid.getDataProvider();
+    }
+
+    private ListDataProvider<Dependency> getDependencyDataProvider() {
+        return new ListDataProvider<>(
+                AppSecService.getInstance().getDependencies());
+    }
+
+    private ListDataProvider<String> getGroupDataProvider() {
+        return new ListDataProvider<>(getDependencyDataProvider().getItems()
+                .stream().map(Dependency::getGroup).filter(Objects::nonNull)
+                .distinct().sorted().collect(Collectors.toList()));
+    }
+
+    private ListDataProvider<Ecosystem> getEcosystemDataProvider() {
+        return new ListDataProvider<>(Arrays.asList(Ecosystem.values()));
+    }
+
+    private ListDataProvider<Boolean> getIsDevelopmentDataProvider() {
+        return new ListDataProvider<>(
+                Arrays.asList(Boolean.TRUE, Boolean.FALSE));
+    }
+
+    private ListDataProvider<SeverityLevel> getSeverityLevelDataProvider() {
+        return new ListDataProvider<>(Arrays.asList(SeverityLevel.values()));
+    }
+
+    private ListDataProvider<String> getRiskScoreDataProvider() {
+        return new ListDataProvider<>(Arrays.asList(">=0", ">=1", ">=2", ">=3",
+                ">=4", ">=5", ">=6", ">=7", ">=8", ">=9", "=10"));
     }
 }

--- a/appsec-kit-starter/src/main/java/com/vaadin/appsec/views/VulnerabilitiesView.java
+++ b/appsec-kit-starter/src/main/java/com/vaadin/appsec/views/VulnerabilitiesView.java
@@ -8,6 +8,7 @@
  */
 package com.vaadin.appsec.views;
 
+import java.util.Arrays;
 import java.util.Set;
 import java.util.stream.Collectors;
 
@@ -107,9 +108,8 @@ public class VulnerabilitiesView extends AbstractAppSecView {
     public void refresh() {
         Set<Vulnerability> selectedItems = grid.getSelectedItems();
         grid.deselectAll();
-        grid.setItems(AppSecService.getInstance().getVulnerabilities());
-        dependency.setItems(getListDataProvider().getItems().stream()
-                .map(Vulnerability::getDependency).collect(Collectors.toSet()));
+        grid.setDataProvider(getVulnerabilityDataProvider());
+        dependency.setDataProvider(getDependencyDataProvider());
         applyFilters();
         selectedItems.forEach(grid::select);
     }
@@ -121,7 +121,7 @@ public class VulnerabilitiesView extends AbstractAppSecView {
 
     private void buildFilters() {
         ecosystem = new ComboBox<>("Ecosystem");
-        ecosystem.setItems(Ecosystem.MAVEN, Ecosystem.NPM);
+        ecosystem.setDataProvider(getEcosystemDataProvider());
         ecosystem.addValueChangeListener(event -> applyFilters());
 
         dependency = new ComboBox<>("Dependency");
@@ -129,27 +129,19 @@ public class VulnerabilitiesView extends AbstractAppSecView {
         dependency.getStyle().set("--vaadin-combo-box-overlay-width", "350px");
 
         vaadinAnalysis = new ComboBox<>("Vaadin analysis");
-        vaadinAnalysis.setItems(AssessmentStatus.TRUE_POSITIVE,
-                AssessmentStatus.FALSE_POSITIVE, AssessmentStatus.UNDER_REVIEW);
+        vaadinAnalysis.setDataProvider(getAssessmentStatusDataProvider());
         vaadinAnalysis.addValueChangeListener(event -> applyFilters());
 
         developerAnalysis = new ComboBox<>("Developer analysis");
-        developerAnalysis.setItems(AppSecData.VulnerabilityStatus.NOT_SET,
-                AppSecData.VulnerabilityStatus.NOT_AFFECTED,
-                AppSecData.VulnerabilityStatus.FALSE_POSITIVE,
-                AppSecData.VulnerabilityStatus.IN_TRIAGE,
-                AppSecData.VulnerabilityStatus.EXPLOITABLE);
+        developerAnalysis.setDataProvider(getVulnerabilityStatusDataProvider());
         developerAnalysis.addValueChangeListener(event -> applyFilters());
 
         severity = new ComboBox<>("Severity");
-        severity.setItems(SeverityLevel.NONE, SeverityLevel.LOW,
-                SeverityLevel.MEDIUM, SeverityLevel.HIGH,
-                SeverityLevel.CRITICAL);
+        severity.setDataProvider(getSeverityLevelDataProvider());
         severity.addValueChangeListener(event -> applyFilters());
 
         riskScore = new ComboBox<>("CVSS score");
-        riskScore.setItems(">=0", ">=1", ">=2", ">=3", ">=4", ">=5", ">=6",
-                ">=7", ">=8", ">=9", "=10");
+        riskScore.setDataProvider(getRiskScoreDataProvider());
         riskScore.addValueChangeListener(event -> applyFilters());
 
         Component filterBar = buildFilterBar(ecosystem, dependency,
@@ -215,5 +207,38 @@ public class VulnerabilitiesView extends AbstractAppSecView {
     @SuppressWarnings("unchecked")
     private ListDataProvider<Vulnerability> getListDataProvider() {
         return (ListDataProvider<Vulnerability>) grid.getDataProvider();
+    }
+
+    private ListDataProvider<Vulnerability> getVulnerabilityDataProvider() {
+        return new ListDataProvider<>(
+                AppSecService.getInstance().getVulnerabilities());
+    }
+
+    private ListDataProvider<Dependency> getDependencyDataProvider() {
+        return new ListDataProvider<>(getVulnerabilityDataProvider().getItems()
+                .stream().map(Vulnerability::getDependency)
+                .collect(Collectors.toSet()));
+    }
+
+    private ListDataProvider<Ecosystem> getEcosystemDataProvider() {
+        return new ListDataProvider<>(Arrays.asList(Ecosystem.values()));
+    }
+
+    private ListDataProvider<AssessmentStatus> getAssessmentStatusDataProvider() {
+        return new ListDataProvider<>(Arrays.asList(AssessmentStatus.values()));
+    }
+
+    private ListDataProvider<AppSecData.VulnerabilityStatus> getVulnerabilityStatusDataProvider() {
+        return new ListDataProvider<>(
+                Arrays.asList(AppSecData.VulnerabilityStatus.values()));
+    }
+
+    private ListDataProvider<SeverityLevel> getSeverityLevelDataProvider() {
+        return new ListDataProvider<>(Arrays.asList(SeverityLevel.values()));
+    }
+
+    private ListDataProvider<String> getRiskScoreDataProvider() {
+        return new ListDataProvider<>(Arrays.asList(">=0", ">=1", ">=2", ">=3",
+                ">=4", ">=5", ">=6", ">=7", ">=8", ">=9", "=10"));
     }
 }

--- a/appsec-kit-starter/src/main/java/com/vaadin/appsec/views/VulnerabilityDetailsView.java
+++ b/appsec-kit-starter/src/main/java/com/vaadin/appsec/views/VulnerabilityDetailsView.java
@@ -10,6 +10,7 @@ package com.vaadin.appsec.views;
 
 import java.text.DateFormat;
 import java.time.Instant;
+import java.util.Arrays;
 import java.util.Date;
 import java.util.List;
 import java.util.Objects;
@@ -46,6 +47,7 @@ import com.vaadin.flow.component.icon.VaadinIcon;
 import com.vaadin.flow.component.orderedlayout.HorizontalLayout;
 import com.vaadin.flow.component.orderedlayout.VerticalLayout;
 import com.vaadin.flow.component.textfield.TextArea;
+import com.vaadin.flow.data.provider.ListDataProvider;
 import com.vaadin.flow.server.Version;
 
 /**
@@ -328,17 +330,18 @@ public class VulnerabilityDetailsView extends VerticalLayout {
 
     private Component buildDeveloperStatus() {
         developerStatus = new ComboBox<>();
-        developerStatus.setItems(AppSecData.VulnerabilityStatus.NOT_SET,
-                AppSecData.VulnerabilityStatus.NOT_AFFECTED,
-                AppSecData.VulnerabilityStatus.FALSE_POSITIVE,
-                AppSecData.VulnerabilityStatus.IN_TRIAGE,
-                AppSecData.VulnerabilityStatus.EXPLOITABLE);
+        developerStatus.setDataProvider(getDeveloperStatusDataProvider());
         developerStatus.setPlaceholder("Select status");
         developerStatus.setWidth(100, Unit.PERCENTAGE);
         developerStatus.setLabel("Vulnerability status");
         developerStatus.setValue(vulnerabilityDTO.getDeveloperStatus());
         developerStatus.addValueChangeListener(this::developerAnalysisChanged);
         return developerStatus;
+    }
+
+    private ListDataProvider<AppSecData.VulnerabilityStatus> getDeveloperStatusDataProvider() {
+        return new ListDataProvider<>(
+                Arrays.asList(AppSecData.VulnerabilityStatus.values()));
     }
 
     private Component buildDeveloperAnalysis() {

